src/main/kotlin/de/bund/digitalservice/useid/widget/WidgetController.kt

```diff
@@ -104,8 +108,8 @@ class WidgetController(
             .build()
     }
 
-    private fun publishMatomoEvent(category: String, action: String, name: String, sessionId: String?) {
-        val matomoEvent = MatomoEvent(this, category, action, name, sessionId)
+    private fun publishMatomoEvent(category: String, action: String, name: String, sessionId: String?, userAgent: String?) {
```

Comment:
See comment above. Either make the `userAgent` optional or use the nullsafe type here.